### ISSUE TITLE
AJAX Examples

### ISSUE THREAD

**Issue by jkachmar** [chexxor/purescript-howto#1]
Would an AJAX example (preferably with `purescript-affjax`, since it's the the de-facto standard in PureScript-land) fall under the scope of this repository?

Working with AJAX and parsing JSON probably requires a bit of prerequisite knowledge of `Aff` and `Generic`s or `purescript-argonaut` (all three of which are _reasonably_ cargo-cultable), so I'm not sure where it would fall under the hierarchy of examples you've outlined in the README.

**Comment by chexxor** [comment]
Yeah, I'd like to have a how-to for that. AJAX/XHR is relatively complicated for a reason, so having explained those bits would be useful. While it's still necessary to show some code examples, the one thing I want to avoid is prescribing a specific library.

Thanks for the idea!

**Comment by chexxor** [comment]
I'll add a "scope" section to the Readme here to help answer future questions of this kind. :+1:

**Comment by chexxor** [comment]
My initial idea of the structure of an AJAX/XHR docpage would be something like:

- What an HTTP request looks like
- Crafting an HTTP request
  - In Browser
  - In Node
- Explaining that it's async, mention Aff or callback style
- Parsing response

Affjax is just one way to make XHR requests. If on Node-side, then the constraints are different, so the solution is different. Node-side can use solutions like [purescript-simple-request](https://github.com/kika/purescript-simple-request) and [purescript-node HTTP.Client](https://github.com/purescript-node/purescript-node-http/blob/master/src/Node/HTTP/Client.purs)
